barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.cpp

```diff
@@ -139,11 +98,13 @@ AggregationObjectIndices create_honk_recursion_constraints(Builder& builder,
         builder.assert_equal(builder.add_variable(1 << log_circuit_size), key_fields[0].witness_index);
         builder.assert_equal(builder.add_variable(input.public_inputs.size()), key_fields[1].witness_index);
         builder.assert_equal(builder.add_variable(UltraFlavor::has_zero_row ? 1 : 0), key_fields[2].witness_index);
-        builder.assert_equal(builder.add_variable(0), key_fields[4].witness_index);
+        builder.assert_equal(builder.add_variable(1), key_fields[4].witness_index);
```

Comment:
this section needs more comments in general oops. This 1 is the "has_recursive_proof" bool which is pretty useless since all circuits have an aggregation object now.